core/ir/disassemble_shared.c

```diff
@@ -749,9 +751,15 @@ internal_opnd_disassemble(char *buf, size_t bufsz, size_t *sofar DR_PARAM_INOUT,
         case IMMED_DOUBLE_kind:
         case PC_kind:
         case FAR_PC_kind: break;
-        case REG_kind:
-            if (!opnd_is_reg_partial(opnd))
+        case REG_kind: {
+            /* We always want to print the operation size for DR_ISA_REGDEPS
+             * instructions.
+             */
+            bool is_global_isa_mode_synthetic =
+                dr_get_isa_mode(get_thread_private_dcontext()) == DR_ISA_REGDEPS;
```

Comment:
Hmm, this overhead may well show up and slow down disassembly.  Don't we already have a dcontext passed in?  Use that instead of calling get_thread_*.  Even then, can we avoid this extra work on every single register by setting use_size_sfx differently in some way?  Hmm.

So for the operation size we have just one size for the largest source size, right?  Should we print that next to the categories instead of on every register?  It might be misleading on every register since that's not what the size means.

Reply:
This check has been removed, I set `use_size_sfx = 0` for regdeps instructions and print only one size (the operation) after the categories if it's not 0 (e.g., `branch` does not have an operation size).